### PR TITLE
Ruby: Use API interface doc as ToC description

### LINKED ISSUE #1689: Ruby: Use service-level comments in doc ToC
### What

The Ruby documentation ToC uses the API description as the summary of each service object. This one-line description confusingly ends up repeated if there are multiple services in a single API. We should instead use service-level comments from the proto if available. (Potentially we should truncate such comments to the first line to prevent too much verbosity.)

### Context
https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1831/files/644abf5c014d13fc9532ecb371b6568d16bf00f6#r150950486

### PR DESCRIPTION
**Pull request by eoogbe**
Closes #1689

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java [hunk 1]
```diff
@@ -44,10 +44,14 @@
 import com.google.protobuf.Api;
 import java.io.File;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class RubyGapicSurfaceDocTransformer implements ModelToViewTransformer {
   private static final String DOC_TEMPLATE_FILENAME = "ruby/message.snip";
 
+  private static final Pattern SENTENCE_PATTERN = Pattern.compile("^([^.]+)\\.(?: |$)");
```
Comment (geigerj) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: Consider handling `\.\w`:

"This is an API for that turns `google.longrunning.Operations` instances into kittens."

File: src/test/java/com/google/api/codegen/testdata/ruby/ruby_doc_library.baseline [hunk 2]
```diff
@@ -1711,7 +1711,7 @@ module Google
       #
       # | Class | Description |
       # | ----- | ----------- |
-      # | [LibraryServiceClient][] | A simple Google Example Library API. |
+      # | [LibraryServiceClient][] | This API represents a simple digital library |
```
Comment (geigerj) [inline src/test/java/com/google/api/codegen/testdata/ruby/ruby_doc_library.baseline]: Nit: Let's end the sentence with a period.

File: src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java [hunk 3]
```diff
@@ -174,6 +182,29 @@ private TocModuleView generateTocModuleView(
...
+        if (dotIndex == line.length() - 1) {
+          builder.append(line);
+          break;
+        }
+
+        if (line.charAt(dotIndex + 1) == ' ') {
+          builder.append(line.substring(0, dotIndex + 1));
+          break;
+        }
+
```
Comment (geigerj) [inline src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java]: Don't you need to find all occurrences of `.` in the line to check if any is followed by whitespace, not just the first one?

**Comment by eoogbe** [comment]
@geigerj PTAL

**Pull request merged**